src/txdb.cpp

```diff
@@ -44,10 +45,21 @@ uint256 CCoinsViewDB::GetBestBlock() const {
     return hashBestChain;
 }
 
+uint256 CCoinsViewDB::GetUptoBlock() const {
+    uint256 hashBestChain;
+    if (!db.Read(DB_BEST_BLOCK_UPTO, hashBestChain))
+        return uint256();
+    return hashBestChain;
+}
+
 bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlock) {
     CDBBatch batch(db);
     size_t count = 0;
     size_t changed = 0;
+    size_t batch_size = (size_t)GetArg("-dbbatchsize", nDefaultDbBatchSize) << 20;
```

Comment:
Well the relevant constraint is the memory usage peak from allocating the batch, which depends on the batch memory usage, not dbcache memory usage. Also, I don't think anyone will need to change this property (except for tests, where it's very useful to get much more frequent partial flushes).